<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>Yun Yan loves RNA so he can have U</title>
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta name="description" content="On-line CV of Yun Yan">
  <meta name="author" content="Yun Yan">

	<!--link rel="stylesheet/less" href="less/bootstrap.less" type="text/css" /-->
	<!--link rel="stylesheet/less" href="less/responsive.less" type="text/css" /-->
	<!--script src="js/less-1.3.3.min.js"></script-->
	<!--append ‘#!watch’ to the browser URL, then refresh the page. -->
	
	<link href="css/bootstrap.min.css" rel="stylesheet">
	<link href="css/style.css" rel="stylesheet">
	<link href="css/font-awesome-4.2.0/css/font-awesome.css" rel="stylesheet">

  <!-- HTML5 shim, for IE6-8 support of HTML5 elements -->
  <!--[if lt IE 9]>
    <script src="js/html5shiv.js"></script>
  <![endif]-->

  <!-- Fav and touch icons -->
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="img/apple-touch-icon-144-precomposed.png">
  <link rel="apple-touch-icon-precomposed" sizes="114x114" href="img/apple-touch-icon-114-precomposed.png">
  <link rel="apple-touch-icon-precomposed" sizes="72x72" href="img/apple-touch-icon-72-precomposed.png">
  <link rel="apple-touch-icon-precomposed" href="img/apple-touch-icon-57-precomposed.png">
  <link rel="shortcut icon" href="img/favicon.png">
  
	<script type="text/javascript" src="js/jquery.min.js"></script>
	<script type="text/javascript" src="js/bootstrap.min.js"></script>
	<script type="text/javascript" src="js/scripts.js"></script>

	<style>

	</style>
</head>

<body>
<br>
<div class="container-fluid">
	<div class="row clearfix">
		<div class="col-md-12 column">
			<!-- side-bar -->
			<div class="col-md-3 column text-center">
				<section>
					<img alt="140x140" src="img/logo.png" height="140" width="140" class="img-thumbnail" />
				</section>

				<section>
					<!-- <hr> -->
					<ul class="nav nav-stacked nav-pills nav-sidebar">
						<li>
							<a href="about.html">About</a>
						</li>
						<li>
							<a href="toolkits.html">Toolkits</a>
						</li>
						<li>
							<a href="publication.html">Research</a>
						</li>
						<li>
							<a href="playground.html">Playground</a>
						</li>
						<li>
							<a href="blog.html">Blog</a>
						</li>

					</ul>
				</section>

			</div>
			<!-- contents -->
			<div class="col-md-9 column">

				<h1>Yun YAN</h1>
				<p><em>Progress in science depends on new techniques, new discoveries, and new ideas, probably in that order.</em><small><a href="http://genomebiology.com/2002/3/9/comment/1013">[1]</a></small></p>
				<hr>

				<section>
					<h3>Identifying binding peaks of RBP from CLIP-seq data</h3>
					<p class="text-center">
					<a href="http://puriney.github.io/CLIPseqToolkit/" target="_blank">
					<img src="http://i.imgur.com/TbUYpLO.png" height="350" />
					</a>
					</p>
					<p><em>Introduction</em>: CLIP-seq can globally report all the binding positions of RNA binding proteins, like PTB, FOX, etc. The key is to identify the sites where proteins and RNA are bound with high affinity and strength, a.k.a peak finding. Lots of Chip-seq tools had been utilized to find peak of CLIP-seq as both Chip-seq and CLIP-seq scan protein-nucleic interaction sites <a href="http://www.ncbi.nlm.nih.gov/pubmed/21994224">[1]</a>. Here represents the bio-informatics Perl script to do the job. It absorbs and incorporates methods at Ago CLIP-seq (by Darnell) <a href="http://www.ncbi.nlm.nih.gov/pubmed/19536157">[2]</a> and modified FDR methods at PTB CLIP-seq (by Xue) <a href="http://www.ncbi.nlm.nih.gov/pubmed/20064465">[3]</a>.

					<p>
					<table class="table">
					  <tr>
					    <td><p class="pull-left"><em>Language</em>: Perl, R.</p></td>
					    <td>
					    <p class="pull-right">
					    <a class="btn btn-sm btn-success" href="http://puriney.github.io/CLIPseqToolkit/" role="button"><i class="fa fa-github-alt"></i>&nbsp;&nbsp;GitHub Repo</a>
					    </p>
					    </td>
					  </tr>
					</table>			
					</p>
				</section>

				<hr>

				<section>
					<h3>Cassette Exon Finder</h3>

					<p class="text-center">
					<a href="http://puriney.github.io/Textbook-Alternative-Exon/">
						<img src="http://www.nature.com/scitable/content/ne0000/ne0000/ne0000/ne0000/14263384/f1_cart_s1_spec1_f2.jpg"  height="350">
					</a>
					<p>
						<p><em>Introduction</em>: Alternative splicing makes main contributions to mammalian trancriptome dynamics, and further proteomic complexity. Cassette exon is the largest category (over 60%) of alternative exon. Cassette exon is related to disease, like SMA (Spinal Muscular Atrophy), a neorodegenerative disorder, which is directly caused by SMN1 deficiency. The inclusion of exon-7 among SMN gene leads to SMN1 transcripts in normal people, while the unexpected exclusion (or skipping) leads to SMN2 which finally results in infantile death. Here represents the scripts to extract exon trios from gene annotation available at UCSC Genome Browser.</p>

						<p>It fetches exon trios to feed downstream scripts <small>(not-included)</small> for exon inclusion ratio quantification and furthermore detecting differential spliced exons via Fisher's exact test.</p>
					</p>

					<div class="row clearfix">
						<div class="col-md-12 column">
							<div class="col-md-6 column">
							<img class="img-responsive" src="https://raw.github.com/Puriney/Textbook-Alternative-Exon/master/figures/cassette_exon_inclustion_ratio%E8%AE%A1%E7%AE%97%E5%85%AC%E5%BC%8F.png">
							</div>
							<div class="col-md-6 column">
							<img class="img-responsive" src="img/differential splicing exon.jpg" />
							</div>
						</div>
					</div>

					<p>
					<table class="table">
					  <tr>
					    <td><p class="pull-left"><em>Language</em>: Perl, Bash</p></td>
					    <td>
					    <p class="pull-right">
					    <a class="btn btn-sm btn-success" href="http://puriney.github.io/Textbook-Alternative-Exon/" role="button"><i class="fa fa-github-alt"></i>&nbsp;&nbsp;GitHub Repo</a>
					    </p>
					    </td>
					  </tr>
					</table>			
					</p>
				</section>
				<hr>
				<section>
					<h3>Plugin for <code>Pysam</code> to decode MD tag</h3>

					<pre><code>&gt;&gt;&gt; ## Demo
&gt;&gt;&gt; demoMD = "31^AT3T14"
&gt;&gt;&gt; ## Demo read.cigar()
&gt;&gt;&gt; cigarList = [(0, 31), (2,2), (3, 94), (0, 18)]
&gt;&gt;&gt; ## cigar list with match/mismatch information: "
&gt;&gt;&gt; moreCigarList = moreCigar(cigarList, demoMD)
&gt;&gt;&gt; print moreCigarList
[(7, 31), (2, 2), (3, 94), (7, 3), (8, 1), (7, 14)]
					</code></pre>

					<p><em>Introduction</em>: It would enable <code>read.cigar</code> in pysam to parse <code>MD</code> tag information saved in bam/sam format. The syntax of pysam could be found <a href="http://www.cgat.org/%7Eandreas/documentation/pysam/api.html?highlight=cigar#pysam.AlignedRead.cigar">here</a> where <code>0</code> denotes <strong>alignment matched</strong> position, <code>7</code> and <code>8</code> are claimed to denote equal and mismatch respectively. Though <code>MD</code> tag is available in bam/sam alignment files and even the identical bases are manually changed to <code>=</code> in cigar string,  the <strong>sequence</strong> mismatched and matched base are still ambiguously marked by <code>0</code>, thus cannot be distinguished, which make it less straightforward to locate base differences, e.g. SNP.</p>

					<p>
					<table class="table">
					  <tr>
					    <td><p class="pull-left"><em>Language</em>: Python</p></td>
					    <td>
					    <p class="pull-right">
					    <a class="btn btn-sm btn-success" href="https://github.com/Puriney/PysamKissesMDtag" role="button"><i class="fa fa-github-alt"></i>&nbsp;&nbsp;GitHub Repo</a>
					    </p>
					    </td>
					  </tr>
					</table>			
					</p>
				</section>
				
			</div>
		</div>
	</div>
</div>
<br>
<br>
</body>
</html>
