<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>Yun Yan loves RNA so he can have U</title>
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta name="description" content="On-line CV of Yun Yan">
  <meta name="author" content="Yun Yan">

	<!--link rel="stylesheet/less" href="less/bootstrap.less" type="text/css" /-->
	<!--link rel="stylesheet/less" href="less/responsive.less" type="text/css" /-->
	<!--script src="js/less-1.3.3.min.js"></script-->
	<!--append ‘#!watch’ to the browser URL, then refresh the page. -->
	
	<link href="css/bootstrap.min.css" rel="stylesheet">
	<link href="css/style.css" rel="stylesheet">
	<link href="css/font-awesome-4.2.0/css/font-awesome.css" rel="stylesheet">

  <!-- HTML5 shim, for IE6-8 support of HTML5 elements -->
  <!--[if lt IE 9]>
    <script src="js/html5shiv.js"></script>
  <![endif]-->

  <!-- Fav and touch icons -->
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="img/apple-touch-icon-144-precomposed.png">
  <link rel="apple-touch-icon-precomposed" sizes="114x114" href="img/apple-touch-icon-114-precomposed.png">
  <link rel="apple-touch-icon-precomposed" sizes="72x72" href="img/apple-touch-icon-72-precomposed.png">
  <link rel="apple-touch-icon-precomposed" href="img/apple-touch-icon-57-precomposed.png">
  <link rel="shortcut icon" href="img/favicon.png">
  
	<script type="text/javascript" src="js/jquery.min.js"></script>
	<script type="text/javascript" src="js/bootstrap.min.js"></script>
	<script type="text/javascript" src="js/scripts.js"></script>

	<style>

	</style>
</head>

<body>
<br>
<div class="container-fluid">
	<div class="row clearfix">
		<div class="col-md-12 column">
			<!-- side-bar -->
			<div class="col-md-3 column text-center">
				<section>
					<img alt="140x140" src="img/logo.png" height="140" width="140" class="img-thumbnail" />
				</section>

				<section>
					<!-- <hr> -->
					<ul class="nav nav-stacked nav-pills nav-sidebar">
						<li>
							<a href="about.html">About</a>
						</li>
						<li>
							<a href="toolkits.html">Toolkits</a>
						</li>
						<li>
							<a href="publication.html">Research</a>
						</li>
						<li>
							<a href="playground.html">Playground</a>
						</li>
						<li>
							<a href="blog.html">Blog</a>
						</li>

					</ul>
				</section>

			</div>
			<!-- contents -->
			<div class="col-md-9 column">

				<h1>Yun YAN</h1>
				<p><i class="fa fa-quote-left"></i>&nbsp;&nbsp;<em>Progress in science depends on new techniques, new discoveries, and new ideas, probably in that order.</em><small><a href="http://genomebiology.com/2002/3/9/comment/1013" target="_blank">[1]</a></small></p>
				<hr>

				<section>
					<h3>ChIPseeker: an R package for ChIP peak Annotation, Comparison and Visualization</h3>
					<p class="text-center">
					<a href="http://bioconductor.org/packages/devel/bioc/vignettes/ChIPseeker/inst/doc/ChIPseeker.html#average-profile-of-chip-peaks-binding-to-tss-region" target="_blank">
					<img src="http://i.imgur.com/kbP66M9.png" height="350" />
					</a>
					</p>
					<p><em>Introduction</em>: <code>ChIPseeker</code>, a popular package of Bioconductor, retrieves the nearest genes around the peak, annotates genomic region of the peak, and uses statstical methods for estimate the significance of overlap among ChIP peak data sets, etc. It is developed and actively maintained by Guangchuang Yu <a href="http://dx.doi.org/10.1093/bioinformatics/btv145">[1]</a> and I am contributor. 
					</p>
					<p>
					My rols in <code>ChIPseeker</code> is to use bootstrap techniques to estimate confidence interval of average ChIP-seq signals within TSS regions, and visualize the results. The official page for <code>ChIPseeker</code> at Bioconductor is <a href="http://www.bioconductor.org/packages/release/bioc/html/ChIPseeker.html" target="_blank">here</a> and the active developing repository is <a href="https://github.com/GuangchuangYu/ChIPseeker" target="_blank">here</a>. 
					</p>

					<p>
					<table class="table">
					  <tr>
					    <td><p class="pull-left"><em>Language</em>: R</p></td>
					    <td>
					    <p class="pull-right">
					    <a class="btn btn-sm btn-success" href="https://github.com/Puriney/ChIPseeker" role="button"><i class="fa fa-github-alt"></i>&nbsp;&nbsp;GitHub Repo</a>
					    </p>
					    </td>
					  </tr>
					</table>			
					</p>					

				</section>

				<hr>
				
				<section>
					<h3>Identifying binding peaks of RBP from CLIP-seq data</h3>
					<p class="text-center">
					<a href="http://puriney.github.io/CLIPseqToolkit/" target="_blank">
					<img src="http://i.imgur.com/TbUYpLO.png" height="350" />
					</a>
					</p>
					<p><em>Introduction</em>: CLIP-seq can globally report all the binding positions of RNA binding proteins, like PTB, FOX, etc. The key is to identify the sites where proteins and RNA are bound with high affinity and strength, a.k.a peak finding. Lots of Chip-seq tools had been utilized to find peak of CLIP-seq as both Chip-seq and CLIP-seq scan protein-nucleic interaction sites <a href="http://www.ncbi.nlm.nih.gov/pubmed/21994224">[1]</a>. Here represents the bio-informatics Perl script to do the job. It absorbs and incorporates methods at Ago CLIP-seq (by Darnell) <a href="http://www.ncbi.nlm.nih.gov/pubmed/19536157">[2]</a> and modified FDR methods at PTB CLIP-seq (by Xue) <a href="http://www.ncbi.nlm.nih.gov/pubmed/20064465">[3]</a>.

					<p>
					<table class="table">
					  <tr>
					    <td><p class="pull-left"><em>Language</em>: Perl, R.</p></td>
					    <td>
					    <p class="pull-right">
					    <a class="btn btn-sm btn-success" href="http://puriney.github.io/CLIPseqToolkit/" role="button"><i class="fa fa-github-alt"></i>&nbsp;&nbsp;GitHub Repo</a>
					    </p>
					    </td>
					  </tr>
					</table>			
					</p>
				</section>

				<hr>

				<section>
					<h3>Cassette Exon Finder</h3>

					<p class="text-center">
					<a href="http://puriney.github.io/Textbook-Alternative-Exon/">
						<img src="http://www.nature.com/scitable/content/ne0000/ne0000/ne0000/ne0000/14263384/f1_cart_s1_spec1_f2.jpg"  height="350">
					</a>
					<p>
						<p><em>Introduction</em>: Alternative splicing makes main contributions to mammalian trancriptome dynamics, and further proteomic complexity. Cassette exon is the largest category (over 60%) of alternative exon. Cassette exon is related to disease, like SMA (Spinal Muscular Atrophy), a neorodegenerative disorder, which is directly caused by SMN1 deficiency. The inclusion of exon-7 among SMN gene leads to SMN1 transcripts in normal people, while the unexpected exclusion (or skipping) leads to SMN2 which finally results in infantile death. Here represents the scripts to extract exon trios from gene annotation available at UCSC Genome Browser.</p>

						<p>It fetches exon trios to feed downstream scripts <small>(not-included)</small> for exon inclusion ratio quantification and furthermore detecting differential spliced exons via Fisher's exact test.</p>
					</p>

					<div class="row clearfix">
						<div class="col-md-12 column">
							<div class="col-md-6 column">
							<img class="img-responsive" src="https://raw.github.com/Puriney/Textbook-Alternative-Exon/master/figures/cassette_exon_inclustion_ratio%E8%AE%A1%E7%AE%97%E5%85%AC%E5%BC%8F.png">
							</div>
							<div class="col-md-6 column">
							<img class="img-responsive" src="img/differential splicing exon.jpg" />
							</div>
						</div>
					</div>

					<p>
					<table class="table">
					  <tr>
					    <td><p class="pull-left"><em>Language</em>: Perl, Bash</p></td>
					    <td>
					    <p class="pull-right">
					    <a class="btn btn-sm btn-success" href="http://puriney.github.io/Textbook-Alternative-Exon/" role="button"><i class="fa fa-github-alt"></i>&nbsp;&nbsp;GitHub Repo</a>
					    </p>
					    </td>
					  </tr>
					</table>			
					</p>
				</section>
				<hr>
				<section>
					<h3>Plugin for <code>Pysam</code> to decode MD tag</h3>

					<pre><code>&gt;&gt;&gt; ## Demo
&gt;&gt;&gt; demoMD = "31^AT3T14"
&gt;&gt;&gt; ## Demo read.cigar()
&gt;&gt;&gt; cigarList = [(0, 31), (2,2), (3, 94), (0, 18)]
&gt;&gt;&gt; ## cigar list with match/mismatch information: "
&gt;&gt;&gt; moreCigarList = moreCigar(cigarList, demoMD)
&gt;&gt;&gt; print moreCigarList
[(7, 31), (2, 2), (3, 94), (7, 3), (8, 1), (7, 14)]
					</code></pre>

					<p><em>Introduction</em>: It would enable <code>read.cigar</code> in pysam to parse <code>MD</code> tag information saved in bam/sam format. The syntax of pysam could be found <a href="http://www.cgat.org/%7Eandreas/documentation/pysam/api.html?highlight=cigar#pysam.AlignedRead.cigar">here</a> where <code>0</code> denotes <strong>alignment matched</strong> position, <code>7</code> and <code>8</code> are claimed to denote equal and mismatch respectively. Though <code>MD</code> tag is available in bam/sam alignment files and even the identical bases are manually changed to <code>=</code> in cigar string,  the <strong>sequence</strong> mismatched and matched base are still ambiguously marked by <code>0</code>, thus cannot be distinguished, which make it less straightforward to locate base differences, e.g. SNP.</p>

					<p>
					<table class="table">
					  <tr>
					    <td><p class="pull-left"><em>Language</em>: Python</p></td>
					    <td>
					    <p class="pull-right">
					    <a class="btn btn-sm btn-success" href="https://github.com/Puriney/PysamKissesMDtag" role="button"><i class="fa fa-github-alt"></i>&nbsp;&nbsp;GitHub Repo</a>
					    </p>
					    </td>
					  </tr>
					</table>			
					</p>
				</section>
				

				<!-- toolkits under construction -->
				<hr><hr>
				<section>
					<!-- <h2>Toolkits under development <i class="fa fa-paragraph"></i><h2> -->
					<!-- <hr> -->
					<h3><code>Minery</code>, a web server for Chemical In Vitro In Vivo Profilling</h3>
					<div class="row">
						<div class="col-md-4 column">
							<a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0099863" target="_blank"><img class="img-responsive img-thumbnail" src="img/ciip.png"></a>
						</div>

						<div class="col-md-8 column">
							<p>CIIP (Chemical In Vitro In Vivo Profilling) is a toolkit with ultimate goal to model the similarities among the bio-activity of small molecules and high-throughput screening data. It is promising for researches in chemeoinformatics and drug discovery.</p>
							<p>Fed with compounds of interest, <code>Minery</code> is dedicated to fetch bioassay data from available database, for example, PubChem.</p>
							<p><code>Minery</code> is implemented in <code>R</code> and deployed at <a href="https://www.shinyapps.io/">Shinyapps</a> thanks to Rstudio.</p>
							<p><em>It will be open-source once it is finished.</em></p>
						</div>
					</div>
					<p>
						<p class="alignleft">
							<button type="button" class="btn btn-warning btn-sm" disabled="disabled"><i class="fa fa-spinner fa-spin"></i>&nbsp;&nbsp;In Progress</button>
						</p>
						<p class="alignright">
							<a class="btn btn-sm btn-success" href="https://yunyan.shinyapps.io/Minery/"><i class="fa fa-cloud"></i>&nbsp;&nbsp;Server</a>
						</p>
					</p>
					<div style="clear: both;"></div>
				</section>
				<hr>
				<section>
					<h3><code>aRtist</code>, repository dedicated to data visualization in bioinformatics.</h3>

					<!-- sample pictures gallery -->
					<p></p> <!-- spacing -->
					<div class="row clearfix">
						<div class="col-md-12 column">
							<div class="carousel slide" id="carousel-729095">
								<ol class="carousel-indicators">
									<li data-slide-to="0" data-target="#carousel-729095">
									</li>
									<li data-slide-to="1" data-target="#carousel-729095">
									</li>
									<li data-slide-to="2" data-target="#carousel-729095" class="active">
									</li>
								</ol>
								<div class="carousel-inner">
									<div class="item MyCarouselImg">
										<img alt="" src="http://41.media.tumblr.com/f400648eeaaf2bc5ee5539b76f42e545/tumblr_n8um5iIuyE1rgemwzo1_1280.png" class="MyCarouselImg" />
										<div class="carousel-caption">
											<p>
												<strong>Multiple Scatter Plot in one canvas</strong>. Each dot has some blur/transparency. Correlation was computed by t.test. Base package could still produce beautiful plots.			
											</p>
										</div>
									</div>
									<div class="item MyCarouselImg">
										<img alt="" src="http://40.media.tumblr.com/784dddf391f17d0d11dbecd4f8e30bd5/tumblr_n41lhsXciC1rgemwzo1_1280.png" class="MyCarouselImg" />
										<div class="carousel-caption">
											<p>Emission probabilities of HMM modeling PTB binding preferences. Trying to reproduce the results shown in Figure1B of paper “De Novo Prediction of PTBP1 Binding and Splicing Targets Reveals Unexpected Features of Its RNA Recognition and Function”</p>
										</div>
									</div>
									<div class="item MyCarouselImg active">
										<img alt="" src="http://36.media.tumblr.com/bf575f0577b00151b1c32b94390cc16a/tumblr_myi0g9EX5b1rgemwzo1_1280.png" class="MyCarouselImg"  />
										<div class="carousel-caption">
											<p>
												Upper-panel plot assumed the coverage for each position follows normal distribution. For better behavior, estimate 95% CI by using bootstrap via package <code>boot</code>. 
											</p>
										</div>
									</div>
								</div> <a class="left carousel-control" href="#carousel-729095" data-slide="prev"><span class="glyphicon glyphicon-chevron-left"></span></a> <a class="right carousel-control" href="#carousel-729095" data-slide="next"><span class="glyphicon glyphicon-chevron-right"></span></a>
							</div>
						</div>
					</div>					
					<!-- sample pictures end -->

					<p> </p>
					<p> </p>

					<p><code>aRtist</code> repository saves reproducible codes for producing elegant and publication level figures in bioinformatics researches. In addition to the above three figures there are more classic cases in my gallery. I have two strong beliefs residing in the <code>aRtist</code>.  
					<li> Data never lies but data visualization probably could.</li> 
					<li>Reviewers are indeed kids who love cartoon with colors.</li></p>

					<p>
						<p class="alignleft">
							<button type="button" class="btn btn-warning btn-sm" disabled="disabled"><i class="fa fa-spinner fa-spin"></i>&nbsp;&nbsp;In Progress</button>
						</p>
						<p class="alignright">
							<a class="btn btn-sm btn-success" href="http://puriney.tumblr.com/"><i class="fa fa-tumblr-square"></i>&nbsp;&nbsp;Gallery</a>
							<a class="btn btn-sm btn-success" href="https://github.com/Puriney/aRtist" role="button"><i class="fa fa-github-alt"></i>&nbsp;&nbsp;Source</a>
						</p>
					</p>
					<div style="clear: both;"></div>

				</section>
				<hr>
				<section>
					<!-- ggHMM -->
					<h3>ggHMM, visualize Hidden Markov Model</h3>
					Using <code>ggplot2</code> and <code>d3.js</code> to implement elegant visualization of Hidden Markov Model with known parameters. 
					<p>
						<p class="alignleft">
							<button type="button" class="btn btn-warning btn-sm" disabled="disabled"><i class="fa fa-spinner fa-spin"></i>&nbsp;&nbsp;Start-Up</button>
						</p>
						<p class="alignright">
							<a class="btn btn-sm btn-success" href="https://github.com/Puriney/ggHMM" role="button"><i class="fa fa-github-alt"></i>&nbsp;&nbsp;Source</a>
						</p>
					</p>
					<div style="clear: both;"></div>
				</section>

								
			</div>
		</div>
	</div>
</div>

<div id="footer">
  <div class="container">
    <hr>

      <p class="pull-left">
      	<a href="about.html"><i class="fa fa-user">&nbsp;&nbsp;</i>About</a>&nbsp;&nbsp;
     	<a href="toolkits.html"><i class="fa fa-rocket"></i>&nbsp;&nbsp;Toolkits</a>&nbsp;&nbsp;
		<a href="publication.html"><i class="fa fa-graduation-cap"></i>&nbsp;&nbsp;</i>Research</a>&nbsp;&nbsp;
		<a href="playground.html"><i class="fa fa-flask">&nbsp;&nbsp;</i>Playground</a>&nbsp;&nbsp;
		<a href="blog.html"><i class="fa fa-lightbulb-o"></i>&nbsp;&nbsp;Blog</a></p>

    <br>
  </div>
</div> 
</body>
</html>
